src/App.js

```diff
@@ -1,9 +1,28 @@
-import React, { useState } from 'react';
+import React from 'react';
 import Router from './core/router';
 import Page from './core/components/Page';
 import getPageLayout from './core/helper/getPageLayout';
+import UserAPIClient from '../src/core/services/UserAPIClient';
+import { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { COMMON_ERROR } from '../src/core/constants';
+import { message } from 'antd';
 
-const App = () => {
+const App = (props) => {
+  useEffect(() => {
+    fetchUser();
+  }, []);
+
+  async function fetchUser () {
+    try {
+      const infoUser = await UserAPIClient.getInfoPlayer();
+      props.setUserName(infoUser.data.username);
```

Comment:
Why you don't update state ```infoUser``` instead?

Reply:
just need to get user's name and id after login to identify in chat box

Reply:
What if in future we need more user info, you have to define new state?

Reply:
Because you have define user model from the first so you don't have to change it. Just remember consider ability of extend features in future to define better model